cpp/include/raft/linalg/normalize.cuh

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "detail/normalize.cuh"
+
+namespace raft {
+namespace linalg {
+
+/**
+ * @brief Divide rows by their L2 norm.
+ *
+ * Note that the implementation is efficient for matrices with a large number of rows, not "thick"
+ * matrices with few long rows.
+ *
+ * @tparam ElementType Input/Output data type
+ * @tparam IndexType Integer type used to for addressing
+ * @param[in] handle raft::handle_t
+ * @param[in] in the input raft::device_matrix_view
+ * @param[out] out the output raft::device_matrix_view
+ */
+template <typename ElementType, typename IndexType>
+void rowNormalize(const raft::handle_t& handle,
```

Comment:
I think, our `clang-format` config would recommend to the snake naming case. Shouldn't we stick to that (as the formal source of ground truth, despite a large fraction of the api being camel at the moment)?

Reply:
> I think, our clang-format config would recommend the snake naming case. Shouldn't we stick to that

I prefer snake case too, I was just being consistent with `rowNorm`. But given all the inconsistencies in the codebase, I'm often clueless about this.